File: Heap/Leetcode/Medium/KthLargest/KthLargest.java
// https://leetcode.com/problems/kth-largest-element-in-an-array/

import java.util.PriorityQueue;

public class KthLargest {
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> pq = new PriorityQueue<>((a,b) -> b-a);

        for(int ele: nums) {
            pq.offer(ele);
        }

        for(int i=1; i<= k-1; i++) {
            pq.poll();
        }
        return pq.poll();
    }
}

File: Heap/Leetcode/Medium/KthLargest/KthLargest.cpp
// https://leetcode.com/problems/kth-largest-element-in-an-array/
#include<bits/stdc++.h>
using namespace std;
int findKthLargest(vector<int> &nums, int k)
{
    priority_queue<int> pq;

    for (int &ele : nums)
    {
        pq.push(ele);
    }

    for (int i = 1; i <= k - 1; i++)
    {
        pq.pop();
    }

    return pq.top();
}
int main(void) {
    
    return 0;
}